### PR TITLE
Push Evaluation Grades to Canvas

### PR DESCRIPTION
**Pull request by wynnset**
- Add button under Evaluation Results to push grades to Canvas
- Export automatically creates a muted assignment in Canvas the first time the grades are pushed for an Evaluation

### CODE REVIEW

File: app/controllers/evaluations_controller.php [hunk 1]
```diff
@@ -348,6 +349,207 @@ function export($type, $id)
...
+                $canvasAssignment = $canvasCourse->getAssignment($this, User::get('id'), $event['canvas_assignment_id']);
+            }
+
+            // Create assignment in Canvas if no assignment already associated or if assignment does not exist in Canvas
+            if (!$canvasAssignment) {
+                $assignmentArgs = array(
+                    'integration_id' => $eventId,
+                    'name' => $event['title'],
+                    'description' => $event['description'],
+                    'published' => $event['is_released'],
```
Comment (kitsook) [inline app/controllers/evaluations_controller.php]: If the instructor tried to push the grades after the iPeer event release date, the Canvas assignment will be created as "unpublished" and grades can't be updated.  Users will see an error message `user not authorized to perform that action`

**Comment by wynnset** [comment]
@kitsook thanks for your thorough review. I have addressed everything. Could you please review again?

**Comment by kitsook** [comment]
Not sure if this is the desired outcome.  If a student is not evaluated (i.e. the score on iPeer is `N/A`), the grade of that student on Canvas will be kept as-is and not removed (say if the grade was manually entered in Canvas before the push).

**Comment by wynnset** [comment]
@kitsook I don't know which part of the logic you are referring to. Grades are always pushed, regardless of whether they are 0 or non-zero. As far as I understand, the only time grades would not get pushed is if they are a "Survey" evaluation, which means it is non-graded. In that case, the whole export fails (so it's not just 1 grade that doesn't get exported). Either way, I don't think we need to worry about that use case because I don't think the instructors will be upgrading grades in Canvas. If you have a different use case / steps to reproduce, I can investigate further.

**Comment by wynnset** [comment]
@xcompass this is ready to be pushed to verf for review. The course-wide grade sync is not yet implemented, but I think we should push this up so the core functionality can be reviewed. Hoping to have the course-wide grade sync implemented soon.

**Comment by xcompass** [comment]
@wynnset could you resolve the conflict? Thanks.

**Pull request merged**